packages/cli/src/config/index.ts

```diff
@@ -1,9 +1,18 @@
+import { StringForUnion } from "../utils/typeUtils.js";
 import { StoreUserConfig, StoreConfig } from "./parseStoreConfig.js";
 import { WorldUserConfig, ResolvedWorldConfig } from "./world/index.js";
 
-export type MUDUserConfig = StoreUserConfig & WorldUserConfig;
+export type MUDUserConfig<EnumNames extends StringForUnion = StringForUnion> = StoreUserConfig<EnumNames> &
+  WorldUserConfig;
 export type MUDConfig = StoreConfig & ResolvedWorldConfig;
 
+/** Type helper for defining MUDUserConfig */
+export function defineMUDUserConfig<EnumNames extends StringForUnion = StringForUnion>(
```

Comment:
nit: would be great if we could just call this `mudConfig({...})`

Reply:
For consistency we should probably capitalize `MUD` here.

Reply:
should `storeConfig` be `StoreConfig` then to consistently capitalize them too?

Reply:
doing a little refactoring PR rn, capitalized func names don't feel good. And with MUDConfig I also get conflicts with types and zod stuff

Reply:
`defineMUDConfig` would be a better alternative, but I'd still prefer `mudConfig` as it's shorter